internal/server/singleprocess/service_ondemand_runner.go

```diff
@@ -0,0 +1,55 @@
+package singleprocess
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+	serverptypes "github.com/hashicorp/waypoint/internal/server/ptypes"
+)
+
+// TODO: test
```

Comment:
Team process question: should we make issues out of these TODOs that are liable to get forgotten? We've got half a dozen or maybe even a full dozen scattered throughout our codebase. @hashicorp/waypoint-core @hashicorp/waypoint-frontend

Reply:
These tests should def be written before merging this PR.

Reply:
Maybe I was too quick to @-mention everyone 😅

Reply:
That's a copypasta comment, i'll check.

Reply:
Yeah, we don't have tests for most of the service files today.

Reply:
Hm, it looks like most of the service files have tests. It shouldn't be too bad to write these, could probably copy the `service_config_test.go` to get started?

Reply:
Ah! Dunno what I was looking at! Regardless, test added!